MINI_2016/Homeworks/Third/VBM/h3.Rmd

```diff
@@ -0,0 +1,103 @@
+
+#Random Forest
+Forest is a set of trees which contain a `vote` allowing us to find an appropriate result. We will test it on different values of correlation between trees to find and optimal one by observing the OOB error. Different correlation can be achieved by selecting random input variables.
+
+In order to do that I will make use of the `randomForest` library.
+```{r, message=FALSE, warning=FALSE, cache=TRUE}
+#install.packages("randomForest")
+library(randomForest)
+```
+
+##Random forest on the same variables
+Firstly we will try to train the forest on the same variables as in the previous section.
+```{r, cache=TRUE, fig.align='center'}
+ffit <- randomForest(Dalc ~ ., data=student_mat, importance = TRUE, prox=TRUE)
+head(predict(ffit, type="prob"))
+plot(ffit, main="Model fit on all variables")
+varImpPlot(ffit, main="Importance Plot")
+MDSplot(ffit, student_mat$Dalc, main="MDS Plot")
+```
+The `MDS (Multidimensional scaling) plot` is used to show the level of similarities between given cases. We can observe that the red, class #1 is widely spread and the remaining classes are clustered around the value of -0.2 of the 1st dimensions and the range of -0.1 to 0.1 for the second dimension.
+
+Based on the importance plot and the variable importance we will select the variables for further testing.
+
+
+##Mean Decrease in Accuracy
+Mean Decrease in Accuracy is a measure of how much a model fit would decrease when a given variable is changed, or removed.
+We will test 3 selected variables that have larger values of `MeanDecreaseAccuracy`.
+```{r, cache=TRUE, fig.align='center'}
+ffit2 <- randomForest(Dalc ~ G1 + sex + failures, data=student_mat, importance = TRUE)
+head(predict(ffit2, type="prob"))
+plot(ffit2, main="Model Fit for variables: G1, sex, failures")
+```
+
+##Mean Decrease in Gini
+Gini is a measure of node purity that is used when calculating splits.
+We will test 3 selected variables that have larger values of `MeanDecreaseGini`.
+```{r, cache=TRUE, fig.align='center'}
+ffit3 <- randomForest(Dalc ~ absences + goout + age, data=student_mat, importance = TRUE)
+head(predict(ffit3, type="prob"))
+plot(ffit3, main="Model Fit for variables: absences, goout, age")
+```
+
+##Randomly chosen variables
+I decided to pick variables not used before in this section.
+```{r, cache=TRUE, fig.align='center'}
+ffit4 <- randomForest(Dalc ~ famsize + traveltime + activities, data=student_mat, importance = TRUE)
+head(predict(ffit4, type="prob"))
+plot(ffit4, main="Model Fit for variables: famsize, traveltime, activities")
+```
+#Performance comparison
+```{r, cache=TRUE, fig.align='center'}
+tab <- table(real = student_mat$Dalc, predicted = predict(dalcTree))
+tab
+sum(diag(tab)) / sum(tab)
```

Comment:
Accuracy is not a good measure in this case. We will learn why it is so. Best regards